modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/manager/UserManagerImpl.java

```diff
@@ -666,6 +675,30 @@ private com.liferay.portal.kernel.model.User _addPortalUser(
 		return portalUser;
 	}
 
+	private void _checkFilterableAttribute(
+			Node node, Set<String> filterableFieldNames)
+		throws BadRequestException {
+
+		if (node != null) {
+			ExpressionNode expressionNode = (ExpressionNode)node;
+			boolean filterable = false;
+
+			for (String field : filterableFieldNames) {
+				if (StringUtil.contains(
+						expressionNode.getAttributeValue(), field,
+						StringPool.COLON)) {
+
+					filterable = true;
```

Comment:
Should we also include a `break` here?  Or would it make sense to completely remove the `boolean filterable` and just `return` if the condition is met?

Reply:
yes, probably adding the break to avoid cycling over the rest of the keys is a good idea.
If the condition is not met over the set, we have to throw the exception